### PR TITLE
fix: types

### PR DESCRIPTION
**Pull request by katherinedavenia**
### What type of PR is this? (_check all applicable_)

- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update
- [ ] Setup
- [ ] Code Refactor
- [ ] Performance Improvements
- [ ] Test
- [x] Build
- [ ] CI
- [ ] Revert

### CODE REVIEW

**Review by gangsararyop — DISMISSED** [review]

File: src/redux/slices/auth.ts [hunk 1]
```diff
@@ -25,9 +25,9 @@ const initialState: InitialState = {
       userId: 0,
       subscriptionId: 0,
       subscriptionSalesId: 0,
-      startDate: '',
-      endDate: '',
-      deletedAt: ''
+      startDate: new Date(),
+      endDate: new Date(),
+      deletedAt: new Date()
```
Comment (shadiqfauzi) [inline src/redux/slices/auth.ts:30]: ini kenapa defaultnya dikasih new date ya?

Comment (katherinedavenia) [reply src/redux/slices/auth.ts:30]: gimana ya mas enaknya, soalnya kalo string kosong kena erorr

Comment (katherinedavenia) [reply src/redux/slices/auth.ts:30]: null juga samaa
Screenshot 2023-08-16 at 10 26 53

Comment (katherinedavenia) [reply src/redux/slices/auth.ts:30]: Screenshot 2023-08-16 at 10 29 40
apa gini aja mas @shadiqfauzi

Comment (shadiqfauzi) [reply src/redux/slices/auth.ts:30]: bolehh, karena agak aneh aja default state reduxnya new date hehe kalau bisa kosong gitu

Comment (shadiqfauzi) [reply src/redux/slices/auth.ts:30]: atau gini bisa ga yah? di type user diganti jadi gini
```
startDate: string | Date
endDate: string | Date
deletedAt: string | Date
```

File: src/features/program/api/useGetSchedules.ts [hunk 2]
```diff
@@ -3,7 +3,10 @@ import { useApiClient } from '~/providers/ApiClientProvider'
 import { Schedule } from '../types/program'
 
 export const useGetSchedules = (
-  scheduleParams: { programHeaderId?: number; startBeforeDate?: string },
+  scheduleParams: {
+    programHeaderId?: number | undefined
+    startBeforeDate?: string | undefined
```
Comment (hiljaker) [inline src/features/program/api/useGetSchedules.ts:8]: harusnya gaperlu pake undefined lg deh karena emang udah optional

**Review by shadiqfauzi — APPROVED** [review]

**Comment by shadiqfauzi** [comment]
build local / `yarn tsc --noEmit` di local udah aman kan yahh?

**Review by Vesals — APPROVED** [review]
roketttttt

**Comment by katherinedavenia** [comment]
> yarn tsc --noEmit

Screenshot 2023-08-16 at 11 22 46

seharusnya aman mass

**Pull request merged**